actors/market/src/lib.rs

```diff
@@ -484,30 +482,66 @@ impl Actor {
             e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to load deal proposals")
         })?;
 
-        let mut weights = Vec::with_capacity(params.sectors.len());
+        let mut sectors_data = Vec::with_capacity(params.sectors.len());
         for sector in params.sectors.iter() {
-            let (deal_weight, verified_deal_weight, deal_space) = validate_and_compute_deal_weight(
+            let sector_size = sector
+                .sector_type
+                .sector_size()
+                .map_err(|e| actor_error!(illegal_argument, "sector size unknown: {}", e))?;
+            validate_deals_for_activation(
                 &proposals,
                 &sector.deal_ids,
                 &miner_addr,
                 sector.sector_expiry,
                 curr_epoch,
+                sector_size,
             )
             .map_err(|e| {
                 e.downcast_default(
                     ExitCode::USR_ILLEGAL_STATE,
                     "failed to validate deal proposals for activation",
                 )
             })?;
-            weights.push(SectorWeights { deal_space, deal_weight, verified_deal_weight });
+
+            let commd = if sector.deal_ids.is_empty() {
```

Comment:
Why copy instead of just calling market actor `compute_data_commitment` to avoid repeating ourselves here?  I don't like how we're duplicating the behavior of an existing method inline.

Reply:
I think we should delete the market method if unused – I might have missed that

Reply:
I will leave it in for now, it might be useful during migration (although probably not).